includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1482,30 +1482,48 @@ public function set_payment_method_title_for_order( $order, $payment_method_type
 	 * @return array Array of keyed metadata values.
 	 */
 	protected function get_metadata_from_order( $order, $payment_type ) {
+		switch ( get_class( $this ) ) {
+			case UPE_Split_Payment_Gateway::class:
+				$gateway_type = 'split_upe';
+				break;
+			case UPE_Payment_Gateway::class:
+				$gateway_type = 'upe';
+				break;
+			default:
+				$gateway_type = 'classic';
+		}
+
 		$name     = sanitize_text_field( $order->get_billing_first_name() ) . ' ' . sanitize_text_field( $order->get_billing_last_name() );
 		$email    = sanitize_email( $order->get_billing_email() );
 		$metadata = [
-			'customer_name'  => $name,
-			'customer_email' => $email,
-			'site_url'       => esc_url( get_site_url() ),
-			'order_id'       => $order->get_id(),
-			'order_number'   => $order->get_order_number(),
-			'order_key'      => $order->get_order_key(),
-			'payment_type'   => $payment_type,
+			'customer_name'        => $name,
+			'customer_email'       => $email,
+			'site_url'             => esc_url( get_site_url() ),
+			'order_id'             => $order->get_id(),
+			'order_number'         => $order->get_order_number(),
+			'order_key'            => $order->get_order_key(),
+			'payment_type'         => $payment_type,
+			'gateway_type'         => $gateway_type,
+			'checkout_type'        => $order->get_created_via(),
+			'client_version'       => WCPAY_VERSION_NUMBER,
+			'subscription_payment' => 'no',
 		];
 
-		// If the order belongs to a WCPay Subscription, set the payment context to 'wcpay_subscription' (this helps with associating which fees belong to orders).
-		if ( 'recurring' === (string) $payment_type && ! $this->is_subscriptions_plugin_active() ) {
-			$subscriptions = wcs_get_subscriptions_for_order( $order, [ 'order_type' => 'any' ] );
-
-			foreach ( $subscriptions as $subscription ) {
-				if ( WC_Payments_Subscription_Service::is_wcpay_subscription( $subscription ) ) {
-					$metadata['payment_context'] = 'wcpay_subscription';
-					break;
+		if ( 'recurring' === (string) $payment_type ) {
+			$subscriptions                    = wcs_get_subscriptions_for_order( $order, [ 'order_type' => 'any' ] );
+			$metadata['subscription_payment'] = count( $subscriptions ) > 1 ? 'renewal' : 'initial';
+			$metadata['payment_context']      = 'regular_subscription';
+
+			// If the order belongs to a WCPay Subscription, set the payment context to 'wcpay_subscription' (this helps with associating which fees belong to orders).
+			if ( $this->is_subscriptions_plugin_active() ) {
```

Comment:
It's important to understand that WCPay/WooPayments subscriptions only work when the Subscriptions plugin is __not__ active. When it is active, it's just regular subscriptions.